### DISCUSSION TITLE
[Feature] WithLink support

### DISCUSSION THREAD

**Discussion by nuskey8** [AnnulusGames/LitMotion#105] [Ideas]
Since AddTo is only for OnDestory, it cannot handle complex requirements; a feature like DOTween's SetLink may be needed.

**Comment by nuskey8** [comment]
The proposed API is as follows: add a WithLink method to the Builder and specify the behavior with an enum called LinkBehaviour. This internally uses the MotionHandleLinker component.

```cs
LMotion.Create(0f, 1f, 10f)
    .WithLink(gameObject, LinkBehaviour.CancelOnDisable)
    .BindToPositionX(transform);
```

**Reply by nuskey8** [reply]
The following issues can be considered during implementation:

## Pause/Resume Motion

DOTween's LinkBehaviour has an option to pause/resume. This can also be done with LitMotion, but pause/resume in LitMotion is represented by PlaybackSpeed. Naming something like "PauseOn-" doesn't seem to fit this library. (Also, the restart function is not implemented in the first place, so it is impossible.)

## WithLink after creation

If you add WithLink to the builder, configuration after creation is not possible. Maybe this should be added to MotionHandle instead of builder.

**Reply by nuskey8** [reply]
Another API idea is to add a LinkBehaviour argument to AddTo. I think this is better from a practical standpoint.

```cs
var handle = LMotion.Create(0f, 1f, 10f).RunWithoutBinding();
handle.AddTo(gameObject, LinkBehaviour.CancelOnDisable);
```

**Reply by nuskey8** [reply]
#106 This PR adds LinkBehaviour options to AddTo. Currently there are only three types, but this seems sufficient for the time being.

**Comment by akeit0** [comment]
Isn't ’Discussions’ the best place to talk about this?

**Reply by nuskey8** [reply]
I have moved this issue to discussions.
